lms/djangoapps/courseware/transformers.py

```diff
@@ -0,0 +1,93 @@
+"""
+Grades Transformer
+"""
+from django.test.client import RequestFactory
+
+from openedx.core.lib.block_structure.transformer import BlockStructureTransformer
+from util.user_utils import SystemUser
+from . import module_render
+
+
+class GradesBlockTransformer(BlockStructureTransformer):
+    """
+    The GradesBlockTransformer collects grading information and stores it on
+    the block structure.
+
+    No runtime transformations are performed.
+
+    The following values are stored as xblock_fields on their respective blocks in the
+    block structure:
+
+        due: (datetime) when the problem is due.
+        graded: (boolean)
+        has_score: (boolean)
+        weight: (numeric)
+
+    Additionally, the following value is calculated and stored as a transformer_block_field
+    for each block:
+
+        max_score: (numeric)
+    """
+    VERSION = 1
+    FIELDS_TO_COLLECT = [u'due', u'graded', u'has_score', u'weight']
```

Comment:
Yes, we can add `format` here as well.
I didn't think of it since it's already being collected by a transformer in the Course Blocks API.
But I agree that this Transformer should specify all fields that are needed, rather than depending on them being collected by another Transformer.